lib/Sema/CSBindings.cpp

```diff
@@ -361,8 +361,7 @@ ConstraintSystem::getPotentialBindings(TypeVariableType *typeVar) {
   assert(!typeVar->getImpl().getFixedType(nullptr) && "has a fixed type");
 
   // Gather the constraints associated with this type variable.
-  SmallVector<Constraint *, 8> constraints;
-  llvm::SmallPtrSet<Constraint *, 4> visitedConstraints;
+  SmallPtrSet<Constraint *, 8> constraints;
```

Comment:
Sets are unordered collections and as a result you have no stable order when iterating over them. This is especially bad with sets of pointers since you're almost certainly guaranteed change in order from run to run.

Reply:
I can switch it to `SmallSetVector` which should have insertion order iteration. Will try that next.

Reply:
Oh it looks like `C++`'s `std::set` is guaranteed to use `std::less` and you're guaranteed an iteration order, but again with pointers as the elements you will not be guaranteed to iterate over the pointed-to elements in the same order from run to run.